apps/explorer/src/components/module/utils.ts

```diff
@@ -54,6 +55,7 @@ export function getNormalizedFunctionParameterTypeDetails(
 	const paramTypeText = normalizedFunctionParameterTypeToString(param, functionTypeArgNames);
 	return {
 		isTxContext: paramTypeText === '0x2::tx_context::TxContext',
+		moveTypeSignature: normalizedTypeToMoveTypeSignature(param),
```

Comment:
I'm not sure if these are the right abstractions.

I added `pureBcsSchemaFromOpenMoveTypeSignatureBody` which will give you a bcs schema instance from an `OpenMoveTypeSignatureBody`, which is the GraphQL equivalent of a `NormalizedMoveType`, and then I added `normalizedTypeToMoveTypeSignature` to convert from `NormalizedMoveType`  to `OpenMoveTypeSignature`.  `NormalizedMoveType` will be goin away as part of deprecating JSON RPC.  The new internal state for txbs will use  `OpenMoveTypeSignature`, and I think having any new methods use the new format makes sense, but GraphQL isn't ready to use in production yet, so we need a way to convert from `NormalizedMoveType`.

I don't know if `normalizedTypeToMoveTypeSignature` and `pureBcsSchemaFromOpenMoveTypeSignatureBody` belong in the SDK, it might be better to just add them to explorer directly if we are fully removing the ability for tx.pure to work without explicitly defining a type

Reply:
Yeah I think moving these to the explorer is the move here.